docs/site/Deploying-to-IBM-Cloud.md

```diff
@@ -0,0 +1,209 @@
+building on the cloud.
+
+#### Updating application
+
+We will use the `cfenv` module to simplify some of the Cloud Foundry related
+operations. Install `cfenv` in the project directory.
+
+```sh
+$ npm i cfenv -S
+```
+
+Update the `index.ts` file to the following to enable service binding.
+
+```ts
+import {TodoListApplication} from './application';
+import {ApplicationConfig} from '@loopback/core';
+const datasourceDb = require('./datasources/db.datasource.json');
+const cfenv = require('cfenv');
+const appEnv = cfenv.getAppEnv();
+// 'myCloudant' is the name of the provisioned Cloudant service
+const dbUrl = appEnv.getServiceURL('myCloudant');
+
+if (!appEnv.isLocal) {
+  datasourceDb.url = dbUrl;
+}
+
+export async function main(options?: ApplicationConfig) {
+  // Set the port assined for the app
+  if (!options) {
+    options = {rest: {port: appEnv.port}};
+  } else {
+    if (!options.rest) options.rest = {};
+    options.rest.port = appEnv.port || options.rest.port;
+  }
+
+  const app = new TodoListApplication(options);
+  await app.boot();
+  await app.start();
+
+  const url = app.restServer.url;
+  console.log(`Server is running at ${url}`);
+  return app;
+}
+```
+
+## Cloud setup
+
+### Database
+
+1.  Provision a Cloudant database service, and name it `myCloudant`.
+2.  Log on to the Cloudant Dashboard and create a database named `todo`.
+
+### Connecting the database to app
+
+Before we can use the provisioned database, we have to connect the database to
+an app on Cloud Foundry.
+
+So, create a
+[Node.js SDK app](https://console.bluemix.net/catalog/starters/sdk-for-nodejs)
+and connect it to the provisioned Cloudant service.
+
+Make a note of the name of the app. For eg: `mylb4app`.
+
+## Deploying the app
+
+### Local
+
+1.  Install all dependencies:
+
+```sh
+npm i
+```
+
+2.  Build the app:
+
+```sh
+npm run build
+```
+
+3.  Ensure Cloudant is running locally.
+4.  Start the app:
+
+```sh
+npm start
+```
+
+5.  Go to http://localhost:3000/swagger-ui to load API Explorer and add a todo
+    or two.
+
+6.  Go to http://localhost:3000/todos to see the todos.
+
+### Cloud
+
+Before we can deploy to IBM Cloud, we need to create two artifacts.
+
+1.  `.cfignore` - not mandatory, prevents `cf` from uploading the listed files
+    to IBM Cloud.
+2.  `manifest.yml` - details about the app.
+
+Create a file named `.cfignore` in the app directory with the following content.
+
+```
+.git/
+node_modules/
+vcap-local.js
+.vscode/
+```
+
+To create the `manifest.yml` file, run `cf create-app-manifest` and specify the
+name of the app that was created earlier `mylb4app`.
+
+```sh
+$ cf create-app-manifest mylb4app -p manifest.yml
+```
+
+Now we are ready to deploy the app to IBM Cloud.
+
+1.  Build the app locally:
+
+```sh
+npm run build
+```
+
+2.  Push the app to Cloud Foundry:
+
+```
+cf push mylb4app
+```
+
+3.  Go to https://mylb4app.eu-gb.mybluemix.net/swagger-ui to load the API
```

Comment:
I have a question regarding a host name and the application nam `mylb4app`. Shouldn't users pick a unique application name? I don't see how the same domain `mylb4app.eu-gb.mybluemix.net` can serve every LB4 user trying out cloud deployment by following this guide.

Am I missing something here?

I think the guide should be more clear about the fact that `mylb4app` is just a placeholder that people need to change to an actual unique application name.

Reply:
I think there's one more glitches here.  After going to that URL, it got redirect to another url that won't work. I have to remove the "8080" port from the redirected url and change to `https`. for example: 
https://loopback.io/api-explorer/?url=https://loopback4-example-shopping.mybluemix.net/openapi.json
instead of the original:
https://loopback.io/api-explorer/?url=http://loopback4-example-shopping.mybluemix.net:8080/openapi.json

Reply:
Good find. This is a bug either in the `rest` package or `loopback.io`. Will investigate further and open an issue accordingly.